lib/msal-custom-auth/src/controller/CustomAuthStandardController.ts

```
@@ -115,17 +107,42 @@ export class CustomAuthStandardController extends StandardController implements
         this.signInClient = interactionClientFactory.create(SignInClient);
         this.signUpClient = interactionClientFactory.create(SignUpClient);
         this.resetPasswordClient = interactionClientFactory.create(ResetPasswordClient);
+        this.cacheClient = interactionClientFactory.create(CustomAuthSilentCacheClient);
     }
 
     /*
      * Gets the current account from the cache.
-     * @param getAccountInputs - Inputs for getting the current cached account
+     * @param accountRetrievalInputs - Inputs for getting the current cached account
      * @returns - A promise that resolves to GetAccountResult
      */
-    async getCurrentAccount(getAccountInputs: GetAccountInputs): Promise<GetAccountResult> {
-        const correlationId = this.getCorrelationId(getAccountInputs);
+    getCurrentAccount(accountRetrievalInputs?: AccountRetrievalInputs): GetAccountResult {
+        try {
+            const correlationId = this.getCorrelationId(accountRetrievalInputs);
 
-        throw new Error(`Method not implemented with Parameter ${correlationId}.`);
+            this.logger.info("Getting current account data.");
+
+            const account = this.cacheClient.getCurrentAccount(accountRetrievalInputs?.username);
+
+            if (account) {
+                this.logger.info("Account data found.");
+
+                return new GetAccountResult(
+                    new CustomAuthAccountData(
+                        account,
+                        this.customAuthConfig,
+                        this.cacheClient,
+                        this.logger,
+                        correlationId,
+                    ),
+                );
+            }
+
+            throw new NoCachedAccountFoundError(correlationId);
+        } catch (error) {
+            this.logger.error(`An error occurred during getting current account: ${error}`);
```

Just an head up here: If the error contains sensitive information and you log it without the proper marking, it can lead to privacy incident

---

This is good callout. How the iOS and Android currently handle this situation? The error here could be anything, so it is hard to mark some specific fields when logging it. Maybe we can log the whole error with LogWithPII.

---

Yes, we log using the function dedicated to "PII" and we mask the error as "EUII". Example:
```
MSALLogger.logPII(
           level: .error,
           context: context,
           format: "attribute_validation_failed received from signup/start request for attributes: \(MSALLogMask.maskEUII(invalidAttributes))"
)
```